UI tests fail to install a static theme

In #6836, we adopted the -dev API instead of -prod in the Disco Pane ui tests. Before that, we only surfaced lightweight themes in the -prod Disco Pane, but -dev lists static themes too.

UI tests fail to install a static theme for some reasons... The theme is successfully installed but the test case does not see the confirmation message and thinks the theme has not been installed...

I tried to use FoxPuppet 1.0.2 to leverage this patch: https://github.com/mozilla/FoxPuppet/pull/209 and the latest geckodriver, but this did not work. It could be an issue in FoxPuppet because installing an extension works (and it is the same popup that is shown).

---

Why do we want to install static themes? We'll migrate LWT to ST on November 19th, so that'd be nice to have ui tests reflecting what's in available to our users.

Driver logs:

````
1541775769524	mozrunner::runner	INFO	Running command: "/Applications/Firefox.app/Contents/MacOS/firefox-bin" "-marionette" "-foreground" "-no-remote" "-profile" "/var/folders/0z/ncskq1p525z_vlw492hdzxjh0000gn/T/rust_mozprofile.DvtHWZTzzyPp"
1541775770126	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid host permission: resource://pdf.js/
1541775770126	addons.webextension.screenshots@mozilla.org	WARN	Loading extension 'screenshots@mozilla.org': Reading manifest: Invalid host permission: about:reader*
1541775771591	Marionette	INFO	Listening on port 55904
1541775771682	Marionette	WARN	TLS certificate errors will be ignored for this session
```

Error:

```
discovery_pane = <pages.discopane.DiscoveryPane object at 0x10790a550>, firefox = <foxpuppet.foxpuppet.FoxPuppet object at 0x10790a3d0>, notifications = <function notifications at 0x1076560c8>

@pytest.mark.nondestructive
def test_theme_installs(discovery_pane, firefox, notifications):
"""Test theme install from discovery pane."""
# We want to install a LWT, not a static theme.
theme = discovery_pane.themes[0]
theme.install()
firefox.browser.wait_for_notification(
> notifications.AddOnInstallConfirmation).install()

tests/ui/tests/test_discopane.py:52:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.tox/discopane-ui-tests/lib/python2.7/site-packages/foxpuppet/windows/browser/window.py:63: in wait_for_notification
message=message)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.support.wait.WebDriverWait (session="0e806b89-2673-d24c-8489-b73dc7b9d22a")>, method = <function <lambda> at 0x107911de8>, message = 'AddOnInstallConfirmation was not shown.'

def until(self, method, message=''):
"""Calls the method provided with the driver as an argument until the \
return value is not False."""
screen = None
stacktrace = None

end_time = time.time() + self._timeout
while True:
try:
value = method(self._driver)
if value:
return value
except self._ignored_exceptions as exc:
screen = getattr(exc, 'screen', None)
stacktrace = getattr(exc, 'stacktrace', None)
time.sleep(self._poll)
if time.time() > end_time:
break
> raise TimeoutException(message, screen, stacktrace)
E       TimeoutException: Message: AddOnInstallConfirmation was not shown.

.tox/discopane-ui-tests/lib/python2.7/site-packages/selenium/webdriver/support/wait.py:80: TimeoutException
------------------------------- pytest-selenium --------------------------------
Driver log: /private/var/folders/0z/ncskq1p525z_vlw492hdzxjh0000gn/T/pytest-of-williamdurand/pytest-15/test_theme_installs0/driver.log URL: https://example.com:4000/en-US/firefox/discovery/pane/64.0/darwin/normal WARNING: Failed to gather log types: Message: GET /session/0e806b89-2673-d24c-8489-b73dc7b9d22a/log/types did not match a known command
```

Screenshot:

![ee8a5d19200009c529a0b9d10b1b273a](https://user-images.githubusercontent.com/217628/48269987-32b82d80-e439-11e8-8ffa-da00290e4257.png)

--- a/tests/ui/requirements/requirements.txt
+++ b/tests/ui/requirements/requirements.txt
@@ -4,7 +4,7 @@ attrs==18.2.0
 certifi==2018.8.24
 chardet==3.0.4
 execnet==1.5.0
-FoxPuppet==1.0.1
+FoxPuppet==1.0.2
 funcsigs==1.0.2
 idna==2.7
 more-itertools==4.3.0

--- a/tests/ui/tests/test_discopane.py
+++ b/tests/ui/tests/test_discopane.py
@@ -44,11 +44,9 @@ def test_addon_installs(discovery_pane, firefox, notifications):
 
 @pytest.mark.nondestructive
 def test_theme_installs(discovery_pane, firefox, notifications):
-    """Test theme install from discovery pane."""
-    # We want to install a LWT, not a static theme.
-    theme = discovery_pane.themes[2]
+    """Test static theme install from discovery pane."""
+    theme = discovery_pane.themes[0]
     theme.install()
-    firefox.browser.wait_for_notification(
-        notifications.AddOnInstallConfirmation).install()
+    # Static themes are installed without any extra confirmation step.
     WebDriverWait(firefox.selenium, timeout=5).until(
         lambda _: theme.is_installed)

